src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunCompilationModuleGroupBase.cs

```diff
@@ -21,10 +22,11 @@ public abstract class ReadyToRunCompilationModuleGroupBase : CompilationModuleGr
         private readonly bool _compileGenericDependenciesFromVersionBubbleModuleSet;
         private readonly bool _isCompositeBuildMode;
         private readonly bool _isInputBubble;
-        private readonly ConcurrentDictionary<TypeDesc, bool> _layoutDependsOnOtherModules = new ConcurrentDictionary<TypeDesc, bool>();
-        private readonly ConcurrentDictionary<TypeDesc, bool> _layoutDependsOnOtherVersionBubbles = new ConcurrentDictionary<TypeDesc, bool>();
+        private readonly ConcurrentDictionary<TypeDesc, CompilationUnitSet> _layoutCompilationUnits = new ConcurrentDictionary<TypeDesc, CompilationUnitSet>();
         private readonly ConcurrentDictionary<TypeDesc, bool> _versionsWithTypeCache = new ConcurrentDictionary<TypeDesc, bool>();
         private readonly ConcurrentDictionary<MethodDesc, bool> _versionsWithMethodCache = new ConcurrentDictionary<MethodDesc, bool>();
+        private readonly Dictionary<ModuleDesc, int> _moduleCompilationUnits = new Dictionary<ModuleDesc, int>();
```

Comment:
any particular reason to not make this Concurrent as well?

Reply:
This is a rarely hit hashtable, a lock should be just as performant and its confusing to write up.